### ISSUE TITLE
Handling slashes in directory names and file names

### ISSUE THREAD

**Issue by joorei** [srikanth-lingala/zip4j#139]
I have to handle ZIP files in which the directory names and file names may or may not contain slash characters. I have no access to the source of the ZIP files and can only assume that they were generated from a non-filesystem (maybe a database).

As an (made up) example at the root of the ZIP file a directory `movies/books` may contain a file `The Hobbit/Der Hobbit.txt`. The example would be stored in the ZIP file as two entries:
* `movies/books` (directory flag `true`)
* `movies/books/The Hobbit/Der Hobbit.txt` (directory flag `false`)

This would normally result in a file `Der Hobbit.txt` nested 3 levels deep in directories when extracted by zip4j (or any other major ZIP software as far as I know) as the slash is interpreted as directory separator.

I wrote an implementation that uses not only the slashes in the entry names but the `directory` flag on ZIP entries as well to reconstruct the original directory structure as a tree. I can then use the tree to extract the data into a filesystem with the trouble-slashes encoded or somewhere else where slashes are supported in file/directory names.

----

Long story short: I [uploaded the implementation](https://github.com/joorei/zip-dir-tree) just in case you deem it relevant for zip4j. I'm not sure myself if it fits into the scope of your library because the use case seems quite special. But if you think it does I would be happy to help adjusting and integrating it into zip4j.

**Comment by srikanth-lingala** [comment]
This is an interesting use case. I will analyse your code, and consider the implications of integrating such feature (which arguably is slightly deviating from zip specs) in zip4j and update this thread. I appreciate that you provided a well documented code to look at.

**Comment by srikanth-lingala** [comment]
After some analysis into your code and the requirement in general, I feel that this cannot be part of the zip4j functionality, and this will open up some more uncertain behaviour on different os. In general, the creators of the zip file should be aware of the complications of having slashes in file names. This leads to unexpected behaviour on different OS's. I will be closing this issue as I feel that this is a rather unique request and I am afraid that integrating this into zip4j might lead to more problems than solutions.

**Comment by joorei** [comment]
After encountering some more "specialties" in the ZIP files I have to read, I have to agree that handling such non standard-compliant cases is better kept separate. I updated the repository with a better approach and restructured some things but the general point remains.

I thank you though for taking the time to look into it, it is very much appreciated!
